theano/compile/ops.py

```diff
@@ -509,7 +509,7 @@ def __init__(self, fn, itypes, otypes, infer_shape):
             self.infer_shape = self._infer_shape
 
     def __eq__(self, other):
-        return (type(self) == type(other) and
+        return (isinstance(self, type(other)) and
```

Comment:
rever this change. this isn't equivalent and this is probably the cause of the crash in travis. We need the old behavior.